pytorch_lightning/trainer/trainer.py

```diff
@@ -849,8 +861,7 @@ def _pre_training_routine(self):
         if self.is_global_zero and self.weights_summary is not None and not self.testing:
             ref_model.summarize(mode=self.weights_summary)
 
-        # restore training and model before hpc is called
-        self.checkpoint_connector.restore()
+        self.checkpoint_connector.resume_end()
```

Comment:
Why is `resume_end` here, where it runs only if training but `resume_start` in `_run` where it runs for all stages?

Reply:
You are right! `resume_start` should be in `Trainer.fit()`. This was also the reason one of the early stopping tests failed. 
To match the behavior on master, `Trainer(resume_from_checkpoint)` applies only to fit currently.